### PR TITLE
Add minimal support for the new `.advancedJoin()` join type in Fluent

### PR DESCRIPTION
**Pull request by gwynne**
Restores compatibility with FluentKit 1.37.0

### CODE REVIEW

**Review by 0xTim — APPROVED** [review]

File: Sources/FluentMongoDriver/MongoDB+Join.swift [hunk 1]
```diff
@@ -95,8 +95,23 @@ extension DatabaseQuery {
                         "$unwind": "$\(alias ?? schema)"
                     ]))
                 case .custom:
-                    fatalError()
+                    throw FluentMongoError.unsupportedJoin
                 }
+            case .advancedJoin(let schema, let space, let alias, let method, let filters) where space == nil && filters.count == 1:
```
Comment (Joannis) [inline Sources/FluentMongoDriver/MongoDB+Join.swift]: One nit: We could replace `let space` with `nil`, removing the need of the `space == nil` clause

File: Sources/FluentMongoDriver/MongoDB+Join.swift [hunk 2]
```diff
@@ -95,8 +95,23 @@ extension DatabaseQuery {
                         "$unwind": "$\(alias ?? schema)"
                     ]))
                 case .custom:
-                    fatalError()
+                    throw FluentMongoError.unsupportedJoin
```
Comment (Joannis) [inline Sources/FluentMongoDriver/MongoDB+Join.swift:98]: Huge plus

**Review by Joannis — APPROVED** [review]

File: Sources/FluentMongoDriver/MongoDB+Join.swift (hunk 1 above)
Comment (gwynne) [reply Sources/FluentMongoDriver/MongoDB+Join.swift]: Fixed 👍

File: Sources/FluentMongoDriver/MongoDB+Join.swift (hunk 2 above)
Comment (gwynne) [reply Sources/FluentMongoDriver/MongoDB+Join.swift:98]: Agreed - too many fatal errors all over Fluent as it is

**Pull request merged**